api/bucket.go

```diff
@@ -190,6 +193,11 @@ func (h *BucketHandler) getDirArchive(ctx *gin.Context, dir string) {
 		return
 	}
 
+	// Indicate empty tar.gz archive with the 204 HTTP code
+	if entriesCount < 1 {
+		ctx.Writer.WriteHeader(http.StatusNoContent)
```

Comment:
Updated with WriteHeader method to set the NoContent status (204), but continue with sending archive data with ctx.Writer.